dashboard/src/components/Catalog/Catalog.tsx

```diff
@@ -124,15 +132,22 @@ function Catalog(props: ICatalogProps) {
   const allRepos = uniq(charts.map(c => c.attributes.repo.name));
   const allProviders = uniq(csvs.map(c => c.spec.provider.name));
   const allCategories = uniq(
-    charts
-      .map(c => categoryToReadable(c.attributes.category))
+    categories
+      .map(c => categoryToReadable(c.name))
       .concat(flatten(csvs.map(c => getOperatorCategories(c)))),
   ).sort();
 
+  const firstUpdate = useRef(true);
   useEffect(() => {
-    fetchCharts(cluster, namespace, repo);
-    getCSVs(cluster, namespace);
-  }, [cluster, namespace, repo, fetchCharts, getCSVs]);
+    if (firstUpdate.current) {
+      // actions when the component is mounted for the first time
+      firstUpdate.current = false;
+      setCurrentRepo("");
+      fetchChartCategories(cluster, namespace);
+      fetchCharts(cluster, namespace, repo);
+      getCSVs(cluster, namespace);
+    }
+  }, [dispatch, getCSVs, fetchCharts, fetchChartCategories, cluster, namespace, currentRepo, repo]);
```

Comment:
I don't understand these changes, can't you just add `fetchChartCategories`? Why is the `firstUpdate` check needed?

Reply:
After looking at it a little bit closer,  `firstUpdate` can be removed and replaced by better handling of the state properties.